portfolio/src/main/webapp/request.js

```diff
@@ -188,7 +188,7 @@ function addScheduledTaskToDom(scheduledTask) {
   priorityText.setAttribute('data-task-priority', taskPriority);
   cardBody.appendChild(priorityText);
 
-  if (!isTaskCompletelyScheduled) {
+  if (!Object.values(isTaskCompletelyScheduled)[0]) {
```

Comment:
This approach seems a bit hacky to me. One approach you can consider is to make the response contain two boolean values.

- Whether the task was scheduled at all (either or partial).
- Whether the task was completely scheduled.

Or, you can define some enum in both Java and Javascript that can represent these values.

ex:
public enum TaskSchedulingResult {
  NOT_SCHEDULED,
  PARTIALLY_SCHEDULED,
  COMPLETELY_SCHEDULED
}

Then you'll have to do some conversion from the optional to to this enum somewhere on the server side.

Reply:
I see. I added an enum in both Java and JavaScript for scheduling completeness status.